indico/modules/events/registration/google_wallet.py

```diff
@@ -0,0 +1,269 @@
+        self.http_client = None
+        self.base_url = 'https://walletobjects.googleapis.com/walletobjects/v1'
+        self.batch_url = 'https://walletobjects.googleapis.com/batch'
+        self.class_url = f'{self.base_url}/eventTicketClass'
+        self.object_url = f'{self.base_url}/eventTicketObject'
+
+        # Set up authenticated client
+        if self.settings:
+            self.settings['google_wallet_application_credentials'] = json.loads(
+                self.settings['google_wallet_application_credentials'])
+            self.auth()
+
+    @staticmethod
+    def get_google_wallet_settings(category):
+        while category:
+            if category.google_wallet_settings:
+                return category.google_wallet_settings
+            else:
+                category = category.parent
+        return {}
+
+    def auth(self):
+        """Create authenticated HTTP client using a service account file."""
+        self.credentials = Credentials.from_service_account_info(
+            self.settings['google_wallet_application_credentials'],
+            scopes=['https://www.googleapis.com/auth/wallet_object.issuer'])
+        self.http_client = AuthorizedSession(self.credentials)
+
+    def create_class_template(self, issuer_id: str, class_suffix: str) -> dict:
+        """ This method will return a dict format ticket class template."""
+
+        if self.event.start_dt_local.date() == self.event.end_dt_local.date():
+            display_start = format_datetime(self.event.start_dt_local, format='medium')
+            display_end = format_time(self.event.end_dt_local.time(), format='short')
+        else:
+            display_start = format_date(self.event.start_dt_local, format='medium')
+            display_end = format_date(self.event.end_dt_local, format='medium')
+        display_data = f'{display_start} -> {display_end}'
+
+        dict_template = {
+            'id': f'{issuer_id}.{class_suffix}',
+            'issuerName': self.settings['google_wallet_issuer_name'],
+            'reviewStatus': 'UNDER_REVIEW',
+            'eventName': {
+                'defaultValue': {
+                    'language': 'en-US',
+                    'value': self.event.title
+                }
+            },
+            'venue': {
+                'name': {
+                    'defaultValue': {
+                        'language': 'en-US',
+                        'value': self.event.venue_name if not self.event.room_name
+                        else f'{self.event.venue_name} - Room {self.event.room_name}'
+                    }
+                },
+                'address': {
+                    'defaultValue': {
+                        'language': 'en-US',
+                        'value': self.event.address
+                    }
+                }
+            },
+            'logo': {
+                'sourceUri': {
+                    'uri': (config.LOGO_URL or (config.IMAGES_BASE_URL + '/logo_indico.png'))
```

Comment:
This fails to create the class for me. I think the url should include the domain

Reply:
Fixed.

Reply:
I think you need something like this:

```python
'sourceUri': {
    'uri': f'{config.BASE_URL}{config.LOGO_URL}' if config.LOGO_URL else
           f'{config.BASE_URL}{config.IMAGES_BASE_URL}/logo_indico.png'
}
```

Otherwise I'm getting 400 Bad Request when creating the class..

Reply:
Sorry, you are right.
I fixed the HeroImage in the class obj and not the logo at class template.

Reply:
Fixed it.